graphein/protein/visualisation.py

```diff
@@ -34,13 +37,215 @@ def plot_pointcloud(mesh: Meshes, title: str = "") -> None:
+        mapping = dict(zip(sorted(node_types), count()))
+        colors = [
+            colour_map(mapping[d[colour_by]] / len(node_types))
+            for n, d in G.nodes(data=True)
+        ]
+
+    return colors
+
+
+def colour_edges(
+    G: nx.Graph,
+    colour_map: matplotlib.colors.ListedColormap,
+    colour_by: str = "kind",
+) -> List[Tuple[float, float, float, float]]:
+    """
+    Computes edge colours based on the kind of bond/interaction.
+    :param G: nx.Graph protein structure graph to compute edge colours from
+    :param colour_map: Colourmap to use
+    :param colour_by: Edge attribute to colour by. Currently only "kind" is supported
+    :return: List of edge colours
+    """
+    if colour_by == "kind":
+        edge_types = set(
+            frozenset(a) for a in nx.get_edge_attributes(G, "kind").values()
+        )
+        mapping = dict(zip(sorted(edge_types), count()))
+        colors = [
+            colour_map(
+                mapping[frozenset(G.edges[i]["kind"])] / len(edge_types)
+            )
+            for i in G.edges()
+        ]
+    return colors
+
+
+def plot_protein_structure_graph(
+    G: nx.Graph,
+    angle: int,
+    figsize: Tuple[int, int] = (10, 7),
+    node_alpha: float = 0.7,
+    node_size_min: float = 20.0,
+    node_size_multiplier: float = 20.0,
+    label_node_ids: bool = True,
+    node_colour_map=plt.cm.plasma,
+    edge_color_map=plt.cm.plasma,
+    colour_nodes_by: str = "degree",
+    colour_edges_by: str = "type",
+    edge_alpha: float = 0.5,
+    out_path: Optional[str] = None,
+    out_format: str = ".png",
+):
+    """
+    Plots protein structure graph in Axes3D.
+    :param G:  nx.Graph Protein Structure graph to plot
+    :param angle:  View angle
+    :param figsize: Size of figure
+    :param node_alpha: Controls node transparency
+    :param node_size_min: Specifies node minimum size
+    :param node_size_multiplier: Scales node size by a constant. Node sizes reflect degree.
+    :param label_node_ids: bool indicating whether or not to plot node_id labels
+    :param node_colour_map: colour map to use for nodes
+    :param edge_color_map: colour map to use for edges
+    :param colour_nodes_by: Specifies how to colour nodes. "degree"m "seq_position" or a node eature
+    :param colour_edges_by: Specifies how to colour edges. Currently only "kind" is supported
+    :param edge_alpha: Controls edge transparency
+    :param out_path: If not none, writes plot to this location
+    :param out_format: Fileformat to use for plot
+    :return:
+    """
+    # Get Node Attributes
+    pos = nx.get_node_attributes(G, "coords")
+
+    # Get node colours
+    node_colors = colour_nodes(
+        G, colour_map=node_colour_map, colour_by=colour_nodes_by
+    )
+    edge_colors = colour_edges(
+        G, colour_map=edge_color_map, colour_by=colour_edges_by
+    )
+
+    # 3D network plot
+    with plt.style.context(("ggplot")):
```

Comment:
Are you sure we should set the style in an opinionated fashion? This could be something that an end-user would want to modify on their own.

Reply:
Added an arg to control this